dev/Repeater/APITests/RepeaterTests.cs

```diff
@@ -168,5 +170,72 @@ public void ValidateGetSetBackground()
                 Verify.AreSame(blueBrush, repeater.Background);
             });
         }
+
+        [TestMethod]
+        public void NestedRepeaterWithDataTemplateScenario()
+        {
+            ItemsRepeater rootRepeater = null;
+            ScrollViewer scrollViewer = null;
+            ManualResetEvent viewChanged = new ManualResetEvent(false);
+            RunOnUIThread.Execute(() =>
+            {
+                var anchorProvider = (ScrollAnchorProvider)XamlReader.Load(
+                  @"<controls:ScrollAnchorProvider Width='400' Height='600'
+                     xmlns='http://schemas.microsoft.com/winfx/2006/xaml/presentation'
+                     xmlns:x='http://schemas.microsoft.com/winfx/2006/xaml'
+                     xmlns:controls='using:Microsoft.UI.Xaml.Controls'>
+                    <controls:ScrollAnchorProvider.Resources>
+                        <DataTemplate x:Key='ItemTemplate' >
+                            <TextBlock Text='{Binding}' />
+                        </DataTemplate>
+                        <DataTemplate x:Key='GroupTemplate'>
+                            <StackPanel>
+                                <TextBlock Text='{Binding}' />
+                                <controls:ItemsRepeater ItemTemplate='{StaticResource ItemTemplate}' ItemsSource='{Binding}' VerticalCacheLength='0'/>
+                            </StackPanel>
+                        </DataTemplate>
+                    </controls:ScrollAnchorProvider.Resources>
+                    <ScrollViewer x:Name='scrollviewer'>
+                        <controls:ItemsRepeater x:Name='rootRepeater' ItemTemplate='{StaticResource GroupTemplate}' VerticalCacheLength='0' />
+                    </ScrollViewer>
+                </controls:ScrollAnchorProvider>");
+
+                rootRepeater = (ItemsRepeater)anchorProvider.FindName("rootRepeater");
+                scrollViewer = (ScrollViewer)anchorProvider.FindName("scrollviewer");
+                scrollViewer.ViewChanged += (sender, args) =>
+                {
+                    if (!args.IsIntermediate)
+                    {
+                        viewChanged.Set();
+                    }
+                };
+
+                var itemsSource = new ObservableCollection<ObservableCollection<int>>();
+                for (int i = 0; i < 100; i++)
+                {
+                    itemsSource.Add(new ObservableCollection<int>(Enumerable.Range(0, 5)));
+                };
+
+                rootRepeater.ItemsSource = itemsSource;
+                Content = anchorProvider;
+            });
+
+            for (int i = 1; i < 10; i++)
+            {
+                IdleSynchronizer.Wait();
+                RunOnUIThread.Execute(() =>
+                {
+                    scrollViewer.ChangeView(null, i * 200, null);
```

Comment:
Maybe give (i * 200) a friendly variable name so it's more obvious what this is doing/validating.